templates/zerver/register.html

```diff
@@ -100,7 +100,7 @@
             <label for="id_team_name" class="inline-block label-title">{{ _('Organization name') }}</label>
 
             <div class="help-box margin-top">
-                {{ _('This can be changed later on the settings page.') }}
+                {{ _('This can always be changed later. Keep it short!') }}
```

Comment:
How about: "Shorter is better than longer. This can be changed later.", and then a tooltip that explains that this is used in notifications for users that enable it.

I'm not sure what to call the "short name" field in the non-subdomains case. I agree it is sort of awkward. Any ideas? One reason we need it is that the Organization Name can be in any language, but the Short Name has to be ascii (it is used in bot email addresses). 

Probably we should at minimum populate it automatically as people type in their organization name, if it's all ascii characters.